tests/Xamarin.Android.Bcl-Tests/Xamarin.Android.Bcl-Tests.targets

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll"   TaskName="Xamarin.Android.BuildTools.PrepTasks.ReplaceFileContents" />
+  <Import Project="..\..\build-tools\mono-runtimes\ProfileAssemblies.projitems" />
+  <Target Name="_AddTestAssemblies"
+      DependsOnTargets="_RemapAssemblies"
+      BeforeTargets="ResolveAssemblyReferences">
+    <ItemGroup>
+      <Reference Include="@(MonoTestAssembly->'$(IntermediateOutputPath)%(Identity)')" />
+    </ItemGroup>
+  </Target>
+  <Target Name="_RemapAssemblies"
+      Inputs="@(MonoTestAssembly->'..\..\bin\$(Configuration)\tests\%(Identity)')"
+      Outputs="@(MonoTestAssembly->'$(IntermediateOutputPath)%(Identity)')">
+    <MakeDir Directories="$(IntermediateOutputPath)" />
+    <ItemGroup>
+      <_Source  Include="@(MonoTestAssembly->'..\..\bin\$(Configuration)\tests\%(Identity)')" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_NUnit>$(XAInstallPrefix)xbuild-frameworks\MonoAndroid\v1.0\Xamarin.Android.NUnitLite.dll</_NUnit>
+    </PropertyGroup>
+    <Exec
+        Command="$(RemapAssemblyRefTool) &quot;%(_Source.Identity)&quot; &quot;$(IntermediateOutputPath)%(_Source.Filename)%(_Source.Extension)&quot; nunitlite &quot;$(_NUnit)&quot;"
+    />
+  </Target>
+  <Target Name="_GenerateApp_cs">
+    <ItemGroup>
+      <Compile Include="$(IntermediateOutputPath)\App.cs" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_Assemblies>@(MonoTestAssembly->'"%(Identity)"', ',  ')</_Assemblies>
+    </PropertyGroup>
+    <ReplaceFileContents
```

Comment:
ReplaceFileContents task tries to delete the destination file even when it doesn't exists and thus failing with an unhandled exception. I think we need to add check with `File.Exists` at https://github.com/xamarin/xamarin-android/blob/master/build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/ReplaceFileContents.cs#L32

Reply:
@radekdoulik: [`File.Delete()`](https://msdn.microsoft.com/en-us/library/system.io.file.delete(v=vs.110).aspx) doesn't throw if a file doesn't exist:

> If the file to be deleted does not exist, no exception is thrown.

The build failure is because the `obj/Debug` directory doesn't exist, and thus it can't create the file `obj/Debug/App.cs`.

Reply:
Yup, I wasn't exact. It was caused by the missing directory (`DirectoryNotFoundException`). It was thrown by the `File.Delete` though.

```
01:51:55 				Error executing task ReplaceFileContents: System.IO.DirectoryNotFoundException: Could not find a part of the path "obj/Debug/App.cs".
01:51:55 				  at System.IO.File.Delete (System.String path) [0x00074] in /Users/builder/data/lanes/5263/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/File.cs:179 
01:51:55 				  at Xamarin.Android.BuildTools.PrepTasks.ReplaceFileContents.Execute () [0x000d9] in /Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/ReplaceFileContents.cs:32```